Fixes #29733 - deprecate foreman_url without arguments

lzap opened the pull request:
Probably cleaner. We do not call foreman_url without arguments anymore
after https://github.com/theforeman/community-templates/pull/570 is
merged.

lzap committed:
Fixes #29733 - deprecate foreman_url without arguments

ekohl reviewed (dismissed):
@tbrisker let's get this in before branching.

ekohl commented:
Looks like tests need to be updated though.

tbrisker commented on lib/foreman/foreman_url_renderer.rb:
```diff
@@ -9,7 +9,12 @@ def default_url_options
     # returns the URL for Foreman based on the required action
-    def foreman_url(action = 'provision', params = {})
+    def foreman_url(action, params = {})
+      if action.nil?
```
If `action` parameter isn't passed, `action` won't be `nil` but rather an error will be thrown due to mismatching number of arguments. you can change the default for `action` to `nil` for this to work.

lzap replied:
Damn yeah. Let's do this and then we can actually change this after this is deprecated. Rebased.

ekohl reviewed (dismissed):
LGTM, let's see if Jenkins agrees :)

tbrisker commented:
@lzap looks like some test are still using the deprecated version ;)

lzap commented:
I have fixed the tests, there was one missed use in JunOS template. I have fixed it here, so next sync should overwrite it. Or if you sync this change https://github.com/theforeman/community-templates/pull/730 then this will have conflict and I can rebase.

tbrisker commented:
I was about to sync templates but look like there was a mistake in  theforeman/community-templates#570 that the snapshot test caught, will cherry-pick after branching when it is fixed.

lzap commented:
Rebased.

tbrisker reviewed (approved):
Thanks @lzap !
